### PR TITLE
Updates to support SDK v2.9-

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: pkgs/test/test/runner/load_suite_test.dart [hunk 1]
```diff
@@ -66,7 +66,7 @@ void main() {
   test('a load test forwards errors and completes LoadSuite.suite to null',
       () async {
     var suite = LoadSuite('name', SuiteConfiguration.empty, suitePlatform, () {
-      fail('error');
+      return fail('error');
```
Comment (kevmoo) [inline pkgs/test/test/runner/load_suite_test.dart:69]: Got an analyzer error saying a return value was required...hrm...

File: pkgs/test_core/lib/src/util/io.dart [hunk 2]
```diff
@@ -43,7 +43,7 @@ final String sdkDir = p.dirname(p.dirname(Platform.resolvedExecutable));
 final OperatingSystem currentOS = (() {
   var name = Platform.operatingSystem;
   var os = OperatingSystem.findByIoName(name);
-  if (os == OperatingSystem.none) return os;
+  if (os != OperatingSystem.none) return os;
```
Comment (grouma) [inline pkgs/test_core/lib/src/util/io.dart:46]: Is this right?

File: pkgs/test_api/lib/src/frontend/future_matchers.dart [hunk 3]
```diff
@@ -66,7 +66,7 @@ class _Completes extends AsyncMatcher {
 
       var buffer = StringBuffer();
       buffer.writeln(indent(prettyPrint(value), first: 'emitted '));
-      if (result!.isNotEmpty) buffer.writeln(indent(result, first: '  which '));
+      if (result.isNotEmpty) buffer.writeln(indent(result, first: '  which '));
```
Comment (natebosch) [inline pkgs/test_api/lib/src/frontend/future_matchers.dart:69]: Do CFE and analyzer match with their flow analysis here?

File: pkgs/test_core/lib/src/util/io.dart (hunk 2 above)
Comment (natebosch) [reply pkgs/test_core/lib/src/util/io.dart:46]: It doesn't look right to me.

Why did this need to change? Was there a static error here?

Comment (kevmoo) [reply pkgs/test_core/lib/src/util/io.dart:46]: If the os is NOT the `none` os, return it.

I think this was a bug in the original migration.

File: pkgs/test_api/lib/src/frontend/future_matchers.dart (hunk 3 above)
Comment (kevmoo) [reply pkgs/test_api/lib/src/frontend/future_matchers.dart:69]: yup! w/ 2.9

File: pkgs/test_api/lib/src/frontend/stream_matcher.dart [hunk 4]
```diff
@@ -179,7 +179,7 @@ class _StreamMatcher extends AsyncMatcher implements StreamMatcher {
       buffer.writeln(indent(eventsString, first: 'emitted '));
       if (result.isNotEmpty) buffer.writeln(indent(result, first: '  which '));
       return buffer.toString().trimRight();
-    }, onError: (error) {
+    }, onError: (Object error) {
```
Comment (kevmoo) [inline pkgs/test_api/lib/src/frontend/stream_matcher.dart:182]: this is weird. You'd think `Object` would be inferred here!

Comment (natebosch) [reply pkgs/test_api/lib/src/frontend/stream_matcher.dart:182]: The `onError` argument has a type of `Function`. In the _ideal_ world we'd have union types so we could type it as `void Function(Object)|void Function(Object, StackTrace)`. For now we don't get _any_ inference for argument types, and not static checking for argument count.

File: pkgs/test_core/lib/src/util/io.dart (hunk 2 above)
Comment (natebosch) [reply pkgs/test_core/lib/src/util/io.dart:46]: hmm, it does look like there was a bug in the migration, but there is _also_ a bug in the non-migrated code.

From what I can tell, `OperatingSystem.findByIoName` never returned null.

We might need to dig a bit deeper on this one.

Comment (natebosch) [reply pkgs/test_core/lib/src/util/io.dart:46]: https://github.com/dart-lang/test/pull/1221

Comment (kevmoo) [reply pkgs/test_core/lib/src/util/io.dart]: resolved

**Comment by kevmoo** [comment]
PTAL!

File: pkgs/test/lib/src/runner/browser/chrome.dart [hunk 5]
```diff
@@ -121,13 +121,16 @@ class Chrome extends Browser {
...
+    if (sourceUri.scheme == 'file') return sourceUri;
+    // If the provided sourceUrl is relative, determine the package path.
     var uri = Uri.parse(script);
     var path = p.join(
         p.joinAll(uri.pathSegments.sublist(1, uri.pathSegments.length - 1)),
         sourceUrl);
     return path.contains('/packages/')
         ? Uri(scheme: 'package', path: path.split('/packages/').last)
-        : Uri.file(p.absolute(path));
+        : null;
```
Comment (grouma) [inline pkgs/test/lib/src/runner/browser/chrome.dart:133]: This doesn't seem right?

Comment (kevmoo) [reply pkgs/test/lib/src/runner/browser/chrome.dart:133]: It's a merge from **your** commit! e3bf6f6

File: pkgs/test_api/lib/src/frontend/stream_matcher.dart (hunk 4 above)
Comment (kevmoo) [reply pkgs/test_api/lib/src/frontend/stream_matcher.dart:182]: yup

File: pkgs/test/lib/src/runner/browser/chrome.dart (hunk 5 above)
Comment (natebosch) [reply pkgs/test/lib/src/runner/browser/chrome.dart:133]: We shouldn't be doing merges from master in PRs we we rebase + squash them...

Comment (kevmoo) [reply pkgs/test/lib/src/runner/browser/chrome.dart:133]: uh...so...land my fixes then do the merge in another PR?

Comment (kevmoo) [reply pkgs/test/lib/src/runner/browser/chrome.dart:133]: @natebosch –  I was trying to bring in your fix

Comment (kevmoo) [reply pkgs/test/lib/src/runner/browser/chrome.dart]: did the merge then added my fixes on top. PTAL

File: pkgs/test/test/runner/load_suite_test.dart (hunk 1 above)
Comment (jakemac53) [reply pkgs/test/test/runner/load_suite_test.dart:69]: file an issue?

**Pull request merged**